scripts/Muon/GUI/MuonAnalysis/loadfile/load_file_presenter.py

```diff
@@ -0,0 +1,132 @@
+from __future__ import (absolute_import, division, print_function)
+
+import copy
+
+from Muon.GUI.Common import thread_model
+import Muon.GUI.Common.muon_file_utils as file_utils
+
+
+class BrowseFileWidgetPresenter(object):
+
+    def __init__(self, view, model):
+        self._view = view
+        self._model = model
+
+        # Whether to allow single or multiple files to be loaded
+        self._multiple_files = True
+
+        self._use_threading = True
+        self._load_thread = None
+
+        self._view.on_browse_clicked(self.on_browse_button_clicked)
+        self._view.on_file_edit_changed(self.handle_file_changed_by_user)
+
+    def show(self):
+        self._view.show()
+
+    def cancel(self):
+        if self._load_thread is not None:
+            self._load_thread.cancel()
+
+    def create_load_thread(self):
+        return thread_model.ThreadModel(self._model)
+
+    def get_filenames_from_user(self):
+        file_filter = file_utils.filter_for_extensions(["nxs"])
+        directory = ""
+        filenames = self._view.show_file_browser_and_return_selection(file_filter, [directory],
+                                                                      multiple_files=self._multiple_files)
+        # validate
+        filenames = file_utils.parse_user_input_to_files(";".join(filenames))
+        filenames = file_utils.remove_duplicated_files_from_list(filenames)
+        return filenames
+
+    def on_browse_button_clicked(self):
+        filenames = self.get_filenames_from_user()
+        filenames = file_utils.remove_duplicated_files_from_list(filenames)
+        if not self._multiple_files and len(filenames) > 1:
+            self._view.warning_popup("Multiple files selected in single file mode")
+            self._view.reset_edit_to_cached_value()
+            return
+        if filenames:
+            self.handle_loading(filenames, self._use_threading)
+
+    def handle_file_changed_by_user(self):
+        user_input = self._view.get_file_edit_text()
+        filenames = file_utils.parse_user_input_to_files(user_input)
+        filenames = file_utils.remove_duplicated_files_from_list(filenames)
+        if not filenames:
+            self._view.reset_edit_to_cached_value()
+            return
+        if not self._multiple_files and len(filenames) > 1:
+            self._view.warning_popup("Multiple files selected in single file mode")
+            self._view.reset_edit_to_cached_value()
+            return
+        self.handle_loading(filenames, self._use_threading)
```

Comment:
why are we passing `self._use_threading` when it can be accessed directly in the `handle_loading` method?

Reply:
Yes this should be changed!

Reply:
Actually this might be better left as it is. My idea here was to allow calling code or testing code to access the `handle_loading` directly and be able to switch threading off.

Reply:
could you put a link to one of the tests please? You probably know where to look and it would take me a while to find it.

Reply:
I think if you look at the `handle_loading` you will see where I was going with this. There always has to be at least one `if else` statement which checks where the `_use_threading` attribute is True/False and calls the appropriate load function.

I chose to split this tiny piece of logic into its own function to reduce the size of the `handle_file_changed_by_user` and `on_browse_button_clicked`, since this is common logic to both functions.

This also makes the code more flexible in the future, allowing us to change the function called by threading/no threading in only a single place in the code.

However, I will remove the usage of the threading argument for the time being as no tests actually use it currently.